erpnext/stock/doctype/warehouse/warehouse.py

```diff
@@ -199,4 +229,42 @@ def get_warehouses_based_on_account(account, company=None):
 		frappe.throw(_("Warehouse not found against the account {0}")
 			.format(account))
 
-	return warehouses
\ No newline at end of file
+	return warehouses
+
+def get_warehouse_list(doctype, txt, filters, limit_start, limit_page_length=20, order_by="modified"):
+	"""Getting list of warehouse and there items details from Warehouse and Bin doctype."""
+	user = frappe.session.user
+	customer = frappe.get_value("User", user, "full_name")
+	flt = {"published": 1}
+	if customer != "Administrator":
```

Comment:
- Technically you need to get the customer to which the user `test@user.com` is linked to in dynamic links.
- The current implementation is not correct, do one thing, from the session user, check the customer it has access to, and then get the warehouses of that customer